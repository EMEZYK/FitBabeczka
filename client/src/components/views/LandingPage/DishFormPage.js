import React from "react";

const DishFormPage = () => {
  return <div>Dish form page</div>;
};

export default DishFormPage;
