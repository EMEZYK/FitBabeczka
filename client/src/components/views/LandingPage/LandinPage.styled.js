import styled from "styled-components";
import { Typography } from "../../ui/Typography/Typography.styled";
import { FlexWrapper } from "../../global-styles/Flex.styled";
import { theme } from "../../../theme/theme";

export const LandingFlexWrapper = styled(FlexWrapper)`
  @media (max-width: 991px) {
    min-height: auto;
  }
`;
export const Header = styled.div`
  display: flex;
  min-height: 55vh;
  width: 100%;
  align-items: center;
  justify-content: right;
  background: url(/photos/landingPhoto.jpg) top;
  background-size: cover;
  background-position: 50% 60%;

  @media (max-width: 992px) {
    flex-direction: column;
    justify-content: flex-end;
    margin: 0 auto;
  }

  .wrapper-about {
    width: 25vw;
    min-height: 55%;
    margin: 0 1rem 0 4.2rem;
    padding: 0.3rem 0.6rem;
    color: #000;
    font-size: clamp(0.5rem, 2vw, 1rem);
    background: rgba(255, 255, 255, 0.6);
    backdrop-filter: blur(8px);
    border: 10px solid hsla(0, 0%, 100%, 0.3);

    @media (max-width: 1280px) {
      margin: 0 0 0 1rem;
    }

    @media (max-width: 992px) {
      width: 100%;
      min-height: auto;
    }
  }
`;

export const IntroductionHeading = styled(Typography)`
  font-size: ${({ fontSize }) =>
    fontSize ? fontSize : theme.fontsize.desktop.xl}px;
  font-size: clamp(1.1rem, 3vw, 1.6rem);

  font-weight: ${({ fontWeight }) =>
    fontWeight ? fontWeight : theme.weight.regular};
  text-align: left;
  margin: 0 0 1rem 0;

  @media (max-width: 992px) {
    text-align: center;
  }
`;

export const ShortDescription = styled(Typography)`
  text-align: left;
  font-size: 1.1rem;

  @media (min-width: 768px) and (max-width: 992px) {
    padding: 0 2rem;
  }
  @media (max-width: 767px) {
    display: none;
  }
`;

export const Heading = styled(Typography)`
  font-size: ${({ fontSize }) =>
    fontSize ? fontSize : theme.fontsize.desktop.xxxl}px;
  font-size: clamp(1.8rem, 2.5vw, 2.4rem);
  letter-spacing: 0.1rem;
  font-weight: ${({ fontWeight }) =>
    fontWeight ? fontWeight : theme.weight.regular};
  padding: 0.4rem 1rem;
  margin: 2rem 0;
  font-family: "Kaushan Script", cursiv;

  @media (max-width: 768px) {
    margin: 4rem 0 0 0;
  }
`;
