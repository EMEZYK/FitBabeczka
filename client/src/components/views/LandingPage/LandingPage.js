import FooterComponent from "../../ui/Footer/Footer";
import CategoriesWrapper from "../../ui/CategoriesWrapper/CategoriesWrapper";
import AllDishes from "../../ui/AllDishes/AllDishes";
import useFetchData from "../../../hooks/fetch-data";
import {
  Header,
  IntroductionHeading,
  ShortDescription,
  Heading,
  LandingFlexWrapper,
} from "./LandinPage.styled";

const LandingPage = ({ categories, categoriesLoadingError }) => {
  const defaultParams = {
    url: "/recipes?limit=8",
    method: "GET",
    headers: {
      accept: "*/*",
    },
  };

  let params = defaultParams;

  const { response, error } = useFetchData({
    url: params.url,
    method: params.method,
  });

  const renderRecipes = () => {
    if (!error) {
      return (
        <AllDishes
          recipes={response.filteredData}
          data-testid="latest-recipes"
        />
      );
    } else if (error) {
      return <div>cannot load recipes</div>;
    }
  };

  return (
    <>
      <LandingFlexWrapper>
        <Header>
          <div className="wrapper-about">
            <IntroductionHeading>
              Sprawdzone przepisy na zdrowe słodkości
            </IntroductionHeading>
            <ShortDescription>
              które zachwycą Cię smakiem i aromatem. Nie rezygnuj przyjemności.
              Wypróbuj moje słodkie wypieki i poczuj radość z pieczenia!
            </ShortDescription>
          </div>
        </Header>
        <CategoriesWrapper
          categories={categories}
          categoriesLoadingError={categoriesLoadingError}
        />
        <Heading>Najnowsze przepisy</Heading>
        {renderRecipes()}
        <FooterComponent />
      </LandingFlexWrapper>
    </>
  );
};

export default LandingPage;
