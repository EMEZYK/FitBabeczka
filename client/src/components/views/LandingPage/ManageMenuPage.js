import React from "react";

const EditMenuPage = () => {
  return <div>Edit dishes page </div>;
};

export default EditMenuPage;
