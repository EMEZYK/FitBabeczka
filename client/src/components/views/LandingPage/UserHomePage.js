import React from "react";

const UserHomePage = () => {
  return <div>User HomePage </div>;
};

export default UserHomePage;
