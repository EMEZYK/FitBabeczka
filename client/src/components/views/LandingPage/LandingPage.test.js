import React from "react";
import { Header } from "./LandinPage.styled";
import * as renderer from "react-test-renderer";
import "jest-styled-components";
import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom";
import LandingPage from "./LandingPage";
import CategoriesWrapperComponent from "../../ui/CategoryWrapper/CategoryWrapper";

describe("LandingPage components", () => {
  afterEach(cleanup);

  test("Should take a snapshot',", () => {
    const { asFragment } = render(<LandingPage />);
    expect(asFragment(<LandingPage />)).toMatchSnapshot();
  });

  test("Renders background image in the header", () => {
    const header = renderer.create(<Header />).toJSON();
    expect(header).toHaveStyleRule(
      "background",
      "url(/photos/landingPhoto.jpg) top"
    );
  });

  test("Categories component renders correctly", () => {
    const categoriesComponent = renderer
      .create(<CategoriesWrapperComponent />)
      .toJSON();
    expect(categoriesComponent).toMatchSnapshot();
  });

  test("Renders newest recipes", () => {
    render(<LandingPage />);
    const latestRecipes = screen.queryByTestId("recipes-container");
    expect(latestRecipes).toBeInTheDocument();
  });
});
