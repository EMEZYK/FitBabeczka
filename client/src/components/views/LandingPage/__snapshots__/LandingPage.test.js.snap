// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage components Categories component renders correctly 1`] = `null`;

exports[`LandingPage components Should take a snapshot', 1`] = `
<DocumentFragment>
  .c3 {
  font-size: clamp (1rem,2vw,3rem);
  font-weight: 400;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 94vw;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F4F1F4;
  margin: 0 auto;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 55vh;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: right;
  -webkit-justify-content: right;
  -ms-flex-pack: right;
  justify-content: right;
  background: url(/photos/landingPhoto.jpg) top;
  background-size: cover;
  background-position: 50% 60%;
}

.c2 .wrapper-about {
  width: 25vw;
  min-height: 55%;
  margin: 0 1rem 0 4.2rem;
  padding: 0.3rem 0.6rem;
  color: #000;
  font-size: clamp(0.5rem,2vw,1rem);
  background: rgba(255,255,255,0.6);
  -webkit-backdrop-filter: blur(8px);
  backdrop-filter: blur(8px);
  border: 10px solid hsla(0,0%,100%,0.3);
}

.c4 {
  font-size: 24px;
  font-size: clamp(1.1rem,3vw,1.6rem);
  font-weight: 400;
  text-align: left;
  margin: 0 0 1rem 0;
}

.c5 {
  text-align: left;
  font-size: 1.1rem;
}

.c6 {
  font-size: 36px;
  font-size: clamp(1.8rem,2.5vw,2.4rem);
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  font-weight: 400;
  padding: 0.4rem 1rem;
  margin: 2rem 0;
  font-family: "Kaushan Script",cursiv;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 10vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #000;
  color: #ffff;
}

.c8 p {
  width: 100%;
  font-size: 0.8rem;
  color: white;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(4,1fr);
  grid-template-rows: auto;
  grid-gap: 4em 2em;
  min-height: 60vh;
  width: 90%;
  justify-items: center;
  margin: 2rem 0 6rem 0;
  grid-auto-flow: dense;
}

@media (max-width:576px) {
  .c3 {
    font-size: 14px;
  }
}

@media (max-width:991px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: 100%;
    height: auto;
    min-height: 100%;
    padding: 0 0 1rem 0;
  }
}

@media (max-width:991px) {
  .c1 {
    min-height: auto;
  }
}

@media (max-width:992px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    margin: 0 auto;
  }
}

@media (max-width:1280px) {
  .c2 .wrapper-about {
    margin: 0 0 0 1rem;
  }
}

@media (max-width:992px) {
  .c2 .wrapper-about {
    width: 100%;
    min-height: auto;
  }
}

@media (max-width:992px) {
  .c4 {
    text-align: center;
  }
}

@media (min-width:768px) and (max-width:992px) {
  .c5 {
    padding: 0 2rem;
  }
}

@media (max-width:767px) {
  .c5 {
    display: none;
  }
}

@media (max-width:768px) {
  .c6 {
    margin: 4rem 0 0 0;
  }
}

@media (max-width:876px) {
  .c8 {
    height: 15vh;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

@media (max-width:876px) {
  .c8 p {
    font-size: 0.8rem;
  }
}

@media (max-width:1440px) {
  .c7 {
    grid-template-columns: repeat(3,1fr);
    width: 100%;
  }
}

@media (max-width:1280px) {
  .c7 {
    grid-template-columns: repeat(2,1fr);
    grid-gap: 4em 1em;
    margin: 0 0 4rem 0;
    padding: 1rem;
  }
}

@media (max-width:599px) {
  .c7 {
    grid-template-columns: repeat(1,1fr);
    grid-gap: 6em;
  }
}

<div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div
        class="wrapper-about"
      >
        <p
          class="c3 c4"
        >
          Sprawdzone przepisy na zdrowe słodkości
        </p>
        <p
          class="c3 c5"
        >
          które zachwycą Cię smakiem i aromatem. Nie rezygnuj przyjemności. Wypróbuj moje słodkie wypieki i poczuj radość z pieczenia!
        </p>
      </div>
    </div>
    <p
      class="c3 c6"
    >
      Najnowsze przepisy
    </p>
    <div
      class="c7"
      data-testid="recipes-container"
    />
    <footer
      class="c8"
      data-testid="footer"
    >
      <p>
        ©2022 Ewelina Mężyk, Wszelkie prawa zastrzeżone
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
